src/sql/src/session/vars.rs

```diff
@@ -1306,6 +1306,10 @@ feature_flags!(
         enable_try_parse_monotonic_iso8601_timestamp,
         "the try_parse_monotonic_iso8601_timestamp function"
     ),
+    (
+        enable_consolidate_after_union_negate,
+        "consolidation after Unions that have a Negated input"
+    ),
```

Comment:
By using `feature_flags!`, you get a flag that is disabled by default. Is the logic here to simplify the implementation and then toggle the flag in LD? If you go that way, I'd suggest being a bit careful to ensure that the flag is always turned on in all of our testing infrastructure. In particular, you could add it to the following:

1. Required: The default configuration given to `mzcompose` should be changed https://github.com/MaterializeInc/materialize/blob/a10ab52231d4df0f8f150cab321fe8139b7ad405/misc/python/materialize/mzcompose/services.py#L43-L60 This will enable the flag in CI.
2. Optional, but recommended: You can also have this as a fixed flag for `sqllogictest`, so that folks will have it on even when running tests locally https://github.com/MaterializeInc/materialize/blob/a10ab52231d4df0f8f150cab321fe8139b7ad405/src/sqllogictest/src/runner.rs#L1085-L1097 This is a last resort for flags that we believe should be turned on in production, so that we do not miss them in local testing while they are not retired.

Reply:
> By using feature_flags!, you get a flag that is disabled by default. Is the logic here to simplify the implementation and then toggle the flag in LD?

Well, if there is some way to specify it already from the code that it should be enabled by default, that might be even better. Maybe I could just flip the boolean and call it `disable_consolidate_after_union_negate`? :D Or is there some better standard way to have it on by default?

> I'd suggest being a bit careful to ensure that the flag is always turned on in all of our testing infrastructure.

Hmm, yes, I'll make sure to have it turned on in mzcompose and all slts!

Reply:
I changed the PR to not use `feature_flags!`, and enabled the flag by default.